azure-stack-go-sdk/azure-stack-go-sdk-profiles.md

```diff
@@ -0,0 +1,302 @@
+
+    ```go
+    package main
+
+    import "github.com/Azure/azure-sdk-for-go/profiles/2017-03-09/network/mgmt/network"
+    ```
+
+6. In your function, create and authenticate a client with a New*Client function call. For example to create a virtual network client:
+
+    ```go
+    package main
+
+    import (
+        "github.com/Azure/azure-sdk-for-go/profiles/2017-03-09/network/mgmt/network"
+        "github.com/Azure/go-autorest/autorest"  
+    ) 
+
+    func main() {
+        vnetClient := network.NewVirtualNetworksClientWithBaseURI("<baseURI>", "<subscriptionID>")
+        vnetClient.Authorizer = autorest.NewBearerAuthorizer(token)
+    }
+    ```
+
+    Set `baseURI` to the `ResourceManagerURL` value used in step 2.
+
+    Set `subscriptionID` to the subscription ID value from step 3.
+
+    To create token see Authentication section below
+
+7. Invoke API methods by using the client that you created in the previous step. For example, to create a virtual network by using the client from previous step:
+
+    ```go
+    package main
+
+    import (
+        "github.com/Azure/azure-sdk-for-go/profiles/2017-03-09/network/mgmt/network"
+        "github.com/Azure/go-autorest/autorest"  
+    ) 
+
+    func main() {
+        vnetClient := network.NewVirtualNetworksClientWithBaseURI("<baseURI>", "<subscriptionID>")
+        vnetClient.Authorizer = autorest.NewBearerAuthorizer(token)
+
+        vnetClient.CreateOrUpdate(...)
+    }
+    ```
+    For a complete example of creating a virtual network on Azure Stack by using Azure Go SDK profile, see Example section below.
+
+# Authentication
+To get the `Authorizer` property from Azure Active Directory using Azure Go SDK, install the Go-AutoRest modules. These modules should have been already installed with the Azure Go SDK installation; if not, install the [authentication package](https://github.com/Azure/go-autorest/tree/master/autorest/adal)
+The `Authorizer` must be set as the "authorizer" property of the resource client. There are different methods to get an Authorizer; for a complete list see [here](https://github.com/Azure/go-autorest/tree/master/autorest/adal#acquire-access-token).
+
+This section presents a common way to get authorizer tokens on Azure Stack by using `client credentials`.
+
+1. If a service principal with owner role on the subscription is available, skip this step. Otherwise create a service principal ([instructions](https://docs.microsoft.com/en-us/azure/azure-stack/azure-stack-create-service-principals)) and assign it an "owner" role scoped to your subscription ([instructions](https://docs.microsoft.com/en-us/azure/azure-stack/azure-stack-create-service-principals#assign-role-to-service-principal)). Save the service principal application ID and secret.
```

Comment:
Change the link text to be the `create a service principal` and remove the paranethical. For `assign it...your subscription` change the following link to appear at the end of the sentence as `To set up a service principal, see [article]`